pkg/container/insert_ordered_map_test.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package container
+
+import (
+	"math/rand/v2"
+	"slices"
+	"testing"
+	"testing/quick"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestInsertOrderedMap_Insert(t *testing.T) {
+	err := quick.Check(func(keys []int) bool {
+		if len(keys) < 1 {
```

Comment:
Should we do a check on empty maps to prevent any possible bounds regressions?

Reply:
I'll tweak it to include empty map checks.